### DISCUSSION TITLE
Contribute asking

### DISCUSSION THREAD

**Discussion by v1r4lv0rt3x** [cagscalclabs/lwip-ce#3] [General]
Hello,
I'm a TI -83 developper and I'd like help you for the project and if possible create with you a web text browser using an api of this program can you answer to me fasly ?

**Comment by alessiodam** [comment]
This looks like a great idea! Currently, lwIP-CE isn't yet a dylib and means your web browser will take at least 200kB, which isn't ideal.

**Reply by v1r4lv0rt3x** [reply]
yes it will be a simple text reader and formular reader without JS, I think that I will use distant servers to make easier some things like graphics or image if i can.

**Reply by alessiodam** [reply]
that really should be nice to have, but I don't really see the point of posting it here? Since lwIP-CE is just a port of lwIP (lightweight IP stack).

**Reply by v1r4lv0rt3x** [reply]
it's because lwip-ce to manage the internet connection and to create the http protocol

**Reply by alessiodam** [reply]
lwIP-CE is a fork of lwIP, you should refer to the lwIP documentation.

**Reply by v1r4lv0rt3x** [reply]
It doesn't seem working for me because when I watch the CI I see [Failure](https://github.com/acagliano/lwip-ce/actions/runs/9441793943/job/26635568652)

**Comment by v1r4lv0rt3x** [comment]
In first i wanted contrib d'or thé CI

**Reply by alessiodam** [reply]
I don't understand anything you've written there.

**Reply by v1r4lv0rt3x** [reply]
Sorry it's my french corrector, I said I wanted to contrib to this repo for the CI.

**Reply by alessiodam** [reply]
The github workflows? I mean if you want you can but on my PR, everything works fine I think

**Reply by v1r4lv0rt3x** [reply]
it's not work perfectlly because there is an error at the end look : [this failure](https://github.com/acagliano/lwip-ce/actions/runs/9441793943/job/26635568652)

**Comment by acagliano** [comment]
> Hello, I'm a TI -83 developper and I'd like help you for the project and if possible create with you a web text browser using an api of this program can you answer to me fasly ?

I'll reply this up here. I appreciate the offer, however you may be misunderstanding the intent of lwIP. lwIP is not a conglomeration of network-capable tools such as browsers and chats and games and what not. That is on the end user to create USING lwIP. lwIP merely provides ECM and NCM drivers for communicating over Ethernet hardware, as well as the link/linkoutput layer, IP layer, and protocol layer. While the project includes some lightweight applications mostly provided as examples by the lwIP devs, my intent is not to include apps in the project. The only other planned feature addition is ALTCP_TLS (merging CryptX into this project), but development is currently paused indefinitely.

If you want to do what you've indicated, you would fork the project, and then write your own browser. You would use the HTTP client API inside the include/apps folder, and then take the response from the HTTP requests you can send with that, processing it, and producing something to render visually using perhaps GraphX or your own graphics system. You would then build it all into an application (until such time as lwIP is available as a dylib).

**Reply by v1r4lv0rt3x** [reply]
Sorry for the delay in response,
I absolutely did not intend for the browser to be included in lwIP-CE but it was while creating this browser that I discovered lwIP-CE and therefore I also wanted to know if someone like you could help me to improve the use of the lwIP-CE API in the browser

**Reply by acagliano** [reply]
> Sorry for the delay in response, I absolutely did not intend for the browser to be included in lwIP-CE but it was while creating this browser that I discovered lwIP-CE and therefore I also wanted to know if someone like you could help me to improve the use of the lwIP-CE API in the browser

It's hard bc we don't really know what your plan is. Text-only browser? Do you want to support media? You can use the http client app to issue requests to any site that doesn't require strict SSL/TLS but then the response will be HTML, CSS, and JS. You'd need to either write interpreters for each, or skip them.
Look into the src/apps/http_client.c/h files and let us know what questions you have.

**Reply by alessiodam** [reply]
I am currently creating an HTTP library, I'll post an update here when it's done.
